bluesky/qt_kicker.py

```diff
@@ -1,35 +1,47 @@
-"""
-A utility that, once imported, kicks the qt event loop at regular
-intervals. This lets qt and asyncio peacefully coexist, and it is 
-necessary for matplotlib < 1.5.0.
-"""
-import asyncio
-import matplotlib.backends.backend_qt5
-from matplotlib.backends.backend_qt5 import _create_qApp
+import sys
 
+_INSTALLED = None
 
-_create_qApp()
-qApp = matplotlib.backends.backend_qt5.qApp
 
-from matplotlib._pylab_helpers import Gcf
+def install_qt_kicker():
+    """Install a periodic callback to integrate qt and asyncio event loops
 
-try:
-    _draw_all = Gcf.draw_all  # mpl version >= 1.5
-except AttributeError:
-    # slower, but backward-compatible
-    def _draw_all():
-        for f_mgr in Gcf.get_all_fig_managers():
-            f_mgr.canvas.draw_idle()
+    If a version of the qt bindings are not already imported, this function
+    will do nothing.
 
+    It is safe to call this function multiple times.
+    """
 
-def _qt_kicker():
-    # The RunEngine Event Loop interferes with the qt event loop. Here we
-    # kick it to keep it going.
-    _draw_all()
+    global _INSTALLED
+    if _INSTALLED is not None:
+        return
+    print("HERE")
+    if not any(p in sys.modules for p in ['PyQt4', 'pyside', 'PyQt5']):
+        return
+    print("ALSO HERE")
+    import asyncio
+    import matplotlib.backends.backend_qt5
+    from matplotlib.backends.backend_qt5 import _create_qApp
+    from matplotlib._pylab_helpers import Gcf
 
-    qApp.processEvents()
-    loop.call_later(0.1, _qt_kicker)
+    _create_qApp()
+    qApp = matplotlib.backends.backend_qt5.qApp
 
+    try:
+        _draw_all = Gcf.draw_all  # mpl version >= 1.5
+    except AttributeError:
+        # slower, but backward-compatible
```

Comment:
I could see a reasonable argument for keeping plotting code in `xray_vision` backwards compatible as that is the project that seems most likely to pick up external collaborators the soonest and is a plotting library.  I can't think of any other library where we would need to particularly care (at least right now) since we ~~control the world!~~ are the only users of the rest of the stack